src/Razor/src/Microsoft.VisualStudio.LanguageServerClient.Razor/DefaultProjectConfigurationFilePathStore.cs

```diff
@@ -0,0 +1,92 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Composition;
+using Microsoft.CodeAnalysis.Razor;
+using Microsoft.CodeAnalysis.Razor.ProjectSystem;
+
+namespace Microsoft.VisualStudio.LanguageServerClient.Razor
+{
+    [Shared]
+    [Export(typeof(ProjectConfigurationFilePathStore))]
+    internal class DefaultProjectConfigurationFilePathStore : ProjectConfigurationFilePathStore
+    {
+        private readonly Dictionary<string, string> _mappings;
+        private readonly object _mappingsLock;
+
+        public override event EventHandler<ProjectConfigurationFilePathChangedEventArgs> Changed;
+
+        [ImportingConstructor]
+        public DefaultProjectConfigurationFilePathStore()
+        {
+            _mappings = new Dictionary<string, string>(FilePathComparer.Instance);
```

Comment:
Any harm of doing this in line with the declaration? Also, is the `[ImportingConstructor]` annotation necessary here?

Reply:
No harm in doing it in the declaration. I just do it this way so it's easier to add onto the class later if needed. Also, the importing constructor isn't needed since we don't have any parameters but again I just do it to make it easier in the future